modules/ting/src/Search/TingSearchRequest.php

```diff
@@ -124,13 +207,121 @@ public function getFacets() {
   }
 
   /**
-   * Sets the list of facets the query should work within.
+   * Get the page the search result should start at.
    *
-   * @param string[] $facets
-   *   The facets.
+   * @return int
+   *   The page-number, defaults to 1.
    */
-  public function setFacets($facets) {
-    $this->facets = $facets;
+  public function getPage() {
+    return $this->page;
+  }
+
+  /**
+   * Sets the page the search result should start at.
+   *
+   * @param int $page
+   *   The page-number, defaults to 1.
+   */
+  public function setPage($page) {
+    $this->page = $page;
+  }
+
+  /**
+   * Get the current sorts.
+   *
+   * @return \Ting\Search\TingSearchSort[]
+   *   The sorts.
+   */
+  public function getSorts() {
+    return $this->sorts;
+  }
+
+  /**
+   * Add a sort based on a field to the query.
+   *
+   * @param string $field
+   *   The field to sort on.
+   *
+   * @param string $direction
+   *   The direction, see TingSearchSort::DIRECTION_*
+   */
+  public function addSort($field, $direction) {
+    $this->sorts[] = new TingSearchSort($field, $direction);
+  }
+
+  /**
+   * Gets a raw-query.
+   *
+   * Beware that this is a query passed in via setRawQuery() and _not_ the
+   * final query represented by the rest of the TingSearchRequest instance.
+   *
+   * @return string
+   *   The query.
+   */
+  public function getRawQuery() {
+    return $this->rawQuery;
+  }
+
+  /**
+   * Sets a raw string to be added to the final query.
+   *
+   * Beware that the query passed in should be compatible with the currently
+   * enabled provider.
+   *
+   * @param string $query
+   *   The query.
+   */
+  public function setRawQuery($query) {
+    $this->rawQuery = $query;
+  }
+
+  /**
+   * Get list of material ids the query is constrained to.
+   *
+   * @return string[]
+   *   The ids.
+   */
+  public function getMaterialFilter() {
+    return $this->materialFilterIds;
+  }
+
+  /**
+   * Set the list of material ids the query is constrained to.
```

Comment:
Do we have any reference to where one might find such an id? A reference to a method on a TingObject perhaps?

Reply:
👍 adding comments.